### ISSUE TITLE
Bad suggestion to `clone_from_slice` from `VecDeque`

### ISSUE THREAD

**Issue by Mirko-von-Leipzig** [rust-lang/rust-clippy#8160]
### Summary

Copying items from `VecDeque` to an array using a for-loop triggers a `manual_memcpy` lint.

The suggested code does not work since `VecDeque` is not trivially sliceable i.e. `&vec_deque[..]` is not valid code.

### Lint Name

manual-memcpy

### Reproducer

I tried this code:

```rust
use std::collections::VecDeque;

fn main() {
    let mut dst = [0; 10];
    let src = (0..dst.len()).collect::<VecDeque<_>>();
    
    for i in 0..dst.len() {
        dst[i] = src[i];
    }
}
```

I saw this happen:

```
warning: it looks like you're manually copying between slices
 --> src/main.rs:7:5
  |
7 | /     for i in 0..dst.len() {
8 | |         dst[i] = src[i];
9 | |     }
  | |_____^ help: try replacing the loop by: `dst.clone_from_slice(&src[..dst.len()]);`
  |
  = note: `#[warn(clippy::manual_memcpy)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_memcpy
```

I expected to see this happen:

Probably no lint? Or one needs to do two separate `clone_from_slice` calls for both parts of the `VecDeque.as_slices()`.

And since `T` is copy it should be suggesting `copy_from_slice`.

### Version

```text
rustc 1.57.0 (f1edd0429 2021-11-29)
binary: rustc
commit-hash: f1edd0429582dd29cccacaf50fd134b05593bd9c
commit-date: 2021-11-29
host: x86_64-unknown-linux-gnu
release: 1.57.0
LLVM version: 13.0.0
```

### Additional Labels

@rustbot label +`I-suggestion-causes-error`
